wiki/History_of_osu!/April_Fools/en.md

```diff
@@ -201,94 +201,104 @@ For the April Fools' day of 2019, there was approximately a 1 in 20 chance of he
+[^lemontree-post-peppy]: [Forum post by peppy (2009-04-01) in "Best of No.1 Hits - Lemon Tree"](https://osu.ppy.sh/community/forums/posts/105679)
+[^lemontree-post-millhioref]: [Beatmap by MillhioreF (2012-08-24) "Best of No.1 Hits - Lemon Tree"](https://osu.ppy.sh/beatmapsets/57878#osu/174267)
+
+[^touhousu-ontheweb]: ["osu.ppy.sh - Changed osu! to touhousu! throughout the website as well as the game." on April Fools' Day On The Web](http://aprilfoolsdayontheweb.com/joke/8120/?size=1)
+[^touhousu-osudev-2021-01-27]: [Discord message by Nivalyx in #osu-wiki in osu!dev (2021-01-27)](https://discord.com/channels/188630481301012481/218677502141399041/804215894762848296)
+[^touhousu-forums]: [Forum thread by rcmero (2010-04-01) "touhousu! - April Fools joke? [Resolved]"](https://osu.ppy.sh/community/forums/topics/27612)
+[^touhousu-forums-2]: [Forum thread by rulingvenus (2010-04-01) "Naked Marisa????"](https://osu.ppy.sh/community/forums/topics/27531)
+
+[^nightcore-wiki]: ["Nightcore (mod)" on osu! wiki](/wiki/Game_modifier/Nightcore)
+[^nightcore-yt]: [YouTube video by Nyaruko (2011-03-31) "When osu! tries to do April Fools"](https://www.youtube.com/watch?v=qD5ep6Fykao)
+[^nightcore-frontpage]: ["osu! — rhythm is just a click away" (2011-04-01) on Wayback Machine](https://web.archive.org/web/20110401175252/http://osu.ppy.sh/)
+
+[^fl-forums]: [Forum post by Melty Bagle (2012-03-31) in "[Archived] 'Flashlight mod' on the site...?"](https://osu.ppy.sh/community/forums/posts/1430529)
+[^fl-osudev-2021-01-28]: [Discord message by TicClick in #osu-wiki in osu!dev (2021-01-28)](https://discord.com/channels/188630481301012481/218677502141399041/804260860331687946)
+[^fl-ontheweb]: ["osu.ppy.sh — 'Flashlight' mode on beatmap search page" on April Fools' Day On The Web](http://aprilfoolsdayontheweb.com/joke/11484/?size=1)
+[^fl-forums-2]: [Forum thread by ----- (2012-03-31) "[Archived] 'flashlight mod' on the site...?"](https://osu.ppy.sh/community/forums/topics/79076)
+[^fl-forums-3]: [Forum post by peppy (2012-04-01) in "[Archived] 'flashlight mod' on the site...?"](https://osu.ppy.sh/community/forums/posts/1433063)
+[^fl-forums-4]: [Forum thread by kreph (2012-03-31) "[Archived] Flashlight bugs the website for some browsers"](https://osu.ppy.sh/community/forums/topics/79077)
+[^fl-osudev-2021-01-29]: [Discord message by spaceman_atlas in #osu-wiki in osu!dev (2021-01-29)](https://discord.com/channels/188630481301012481/218677502141399041/804814051209117696)
+
+[^bad-apple-chart]: [Bad Apple Ranking Chart! (2012-04-04)](https://osu.ppy.sh/rankings/osu/charts?spotlight=50)
+[^bad-apple-news]: [News post by Cyclone (2012-04-01) "Bad Apple!! Ranking Chart"](https://osu.ppy.sh/community/forums/topics/79128)
+[^bad-apple-osudev-2021-01-29]: [Discord message by TicClick in #osu-wiki in osu!dev (2021-01-29)](https://discord.com/channels/188630481301012481/218677502141399041/804797922021081138)
```

Comment:
this is just a copy-paste of ^bad-apple-news above and the forum thread is now accessible anyway